pkgs/by-name/ma/manticoresearch/columnar.nix

```diff
@@ -0,0 +1,160 @@
+  };
+  hnswlib = stdenv.mkDerivation (finalAttrs: {
+    pname = "hnswlib";
+    version = "d7bb3bb"; # see HNSW_GITHUB in columnar's cmake/GetHNSW.cmake
+    src = fetchzip {
+      url = "https://github.com/manticoresoftware/hnswlib/archive/d7bb3bb.zip";
+      hash = "sha256-SDYf8J4auzqGBHqFMDFR6TVyNthCA4q6J4glcIssAJU=";
+    };
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+    ];
+    meta = {
+      description = "Header-only C++/python library for fast approximate nearest neighbors";
+      homepage = "https://github.com/nmslib/hnswlib";
+      license = lib.licenses.asl20;
+      platforms = lib.platforms.all;
+    };
+  });
+  pgm-index = stdenv.mkDerivation (finalAttrs: {
+    pname = "pgm-index";
+    version = "pgm_2022_08_02"; # see PGM_GITHUB in columnar's cmake/GetPGM.cmake
+    src = fetchzip {
+      url = "https://github.com/manticoresoftware/PGM-index/archive/refs/heads/pgm_2022_08_02.zip";
+      hash = "sha256-p8CjY9r1TEi91HBac9rZ39Yae2XuNh2yQWbo65n90jU=";
+    };
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+    ];
+    # Use the custom CMakeLists.txt from columnar
+    postPatch = ''
+      cp ${columnarSource}/pgm/CMakeLists.txt CMakeLists.txt
+    '';
+    meta = {
+      description = "Piecewise Geometric Model index - a fast and space-efficient data structure";
+      homepage = "https://github.com/gvinciguerra/PGM-index";
+      license = lib.licenses.asl20;
+      platforms = lib.platforms.all;
+    };
+  });
+  streamvbyte = stdenv.mkDerivation (finalAttrs: {
+    pname = "streamvbyte";
+    version = "efdd9dac"; # see SVB_GITHUB in columnar's cmake/GetStreamvbyte.cmake
+    src = fetchzip {
+      url = "https://github.com/manticoresoftware/streamvbyte/archive/efdd9dac.zip";
+      hash = "sha256-a9E1aWBY/P7wI+kgHqhEiD3THctFfeFcy658RcNpHfQ=";
+    };
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+    ];
+    # Enable SSE4.1 on x86_64 to avoid GCC 14 target attribute issues
+    env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isx86_64 "-msse4.1";
+    # Use the custom CMakeLists.txt from columnar
+    postPatch = ''
+      cp ${columnarSource}/streamvbyte/CMakeLists.txt CMakeLists.txt
+    '';
+    meta = {
+      description = "Fast integer compression library using StreamVByte encoding";
+      homepage = "https://github.com/manticoresoftware/streamvbyte/";
+      license = lib.licenses.asl20;
+      platforms = lib.platforms.all;
+    };
+  });
+  fastpfor = stdenv.mkDerivation (finalAttrs: {
+    pname = "fastpfor";
+    version = "simde"; # see FP_GITHUB in columnar's cmake/GetFastPFor.cmake
+    src = fetchzip {
+      url = "https://github.com/manticoresoftware/FastPFor/archive/refs/heads/simde.zip";
+      hash = "sha256-O9nqgsZKT+q9dHdh2RahF1U8KXCmDp0ld7hNvVREM8k=";
+    };
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+    ];
+    buildInputs = [ simde ];
+    cmakeFlags = [ "-DCMAKE_POLICY_VERSION_MINIMUM=3.5" ];
```

Comment:
This should be patched instead, so that when it is not longer necessary, it can be easily removed